projects/igniteui-angular/src/lib/core/styles/components/query-builder/_query-builder-theme.scss

```diff
@@ -103,48 +117,154 @@
+                 --sb-size: #{rem(10px)};
+
+                max-height: rem(570px);
+                overflow-y: auto;
+                overflow-x: hidden;
+                padding-inline-end: rem(16px);
+            }
+        }
+    }
+
+    %filter-tree__subquery {
+        max-width: rem(960px);
+
+        // Add styles to the subquery itself only if there is a direct .igx-filter-tree__inputs inside it
+        &:has( > %filter-tree__inputs) {
+            background: var-get($theme, 'subquery-header-background');
+            border: rem(1px) solid var-get($theme, 'subquery-border-color');
+            border-radius: var-get($theme, 'subquery-border-radius');
+        }
+
+        // Hide the subquery itself if there is a tree with display none inside
+        &:has( > %query-builder-tree[style='display: none;']) {
+            display: none;
+        }
+
+        > %filter-tree__inputs {
+            padding: rem(12px);
+            border-radius: inherit;
 
+            > igx-input-group,
+            > igx-date-picker,
+            > igx-time-picker {
+                flex-grow: 1;
+            }
+        }
+
+        %filter-tree__buttons {
+            margin-block-start: rem(8px);
+        }
+
+        %query-builder-tree {
+            border-block-start: rem(1px) dashed var-get($theme, 'separator-color');
+            padding: rem(12px);
+        }
+
+        &:empty {
+            display: none;
+        }
+
+    }
+
+    %filter-tree-details-button {
+        margin-inline-start: auto;
+    }
+
+    %query-builder-dialog {
+        display: flex;
+        flex-direction: column;
+        gap: rem(16px);
+        max-width: rem(310px);
+
+        > * {
+            margin: 0 !important;
+        }
+    }
+
+    %advanced-filter__header {
         display: flex;
         align-items: center;
         background-color: var-get($theme, 'header-background');
         color: var-get($theme, 'header-foreground');
         user-select: none;
+        border-radius: $border-radius $border-radius 0 0;
         margin-bottom: 0;
         border-block-end: rem(1px) solid var-get($theme, 'header-border');
+
+        padding-inline: if($variant != 'bootstrap', rem(24px), rem(16px));
+        padding-block: if($variant != 'bootstrap', rem(24px), rem(16px)) rem(16px);
     }
 
-    %advanced-filter__main {
-        display: block;
-        overflow: auto;
-        min-height: pad(rem(138px), rem(164px), rem(214px));
-        max-height: pad(rem(305px), rem(394px), rem(468px));
+    %advanced-filter__title {
+        @if $variant == 'material' {
+            @include type-style('h6');
+        }
 
-        [igxButton] + [igxButton] {
-            margin-inline-start: rem(8px);
+        @if $variant == 'fluent' {
+            @include type-style('h6');
+        }
+
+        @if $variant == 'indigo' {
+            @include type-style('h5');
         }
+
+        @if $variant == 'bootstrap' {
+            @include type-style('h5');
+        }
```

Comment:
Shouldn't these be a part of the typography mixin?

Reply:
if I add them to the typography mixin, we have to remove them after the refactoring for the scoped themes.